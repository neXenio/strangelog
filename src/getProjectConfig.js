// @flow

import { resolve } from 'path';

import jsYaml from 'js-yaml';
import { existsSync, readFileSync } from 'fs-extra';

import type { ConfigType } from './types';

export default function getProjectConfig(): ConfigType {
  const configFilePath = resolve('./.strangelogrc');
  const config = existsSync(configFilePath)
    ? jsYaml.safeLoad(readFileSync(configFilePath).toString())
    : {};

  return {
    path: './changelog',
    components: {},
    ...config
  };
}