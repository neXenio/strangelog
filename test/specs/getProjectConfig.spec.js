// @flow

import { resolve } from 'path';

import { removeSync } from 'fs-extra';

import getProjectConfig from '../../src/getProjectConfig';
import { createTestProject } from '../factories/testProject';

describe('getProjectConfig', () => {

  let testProject, cwd;

  beforeEach(() => {
    cwd = process.cwd();
    testProject = createTestProject();
    process.chdir(resolve(testProject.rootPath));
  });

  afterEach(() => {
    removeSync(testProject.rootPath);
    process.chdir(cwd);
  });

  describe('when there is no .strangelogrc', () => {

    it('returns default configuration', () => {
      removeSync(testProject.configFilePath);
      expect(
        getProjectConfig()
      ).toMatchSnapshot();
    });

  });

  describe('when there is a .strangelogrc', () => {

    it('returns configuration from .strangelogrc merged over defaults', () => {
      expect(
        getProjectConfig()
      ).toMatchSnapshot();
    });

  });

});
