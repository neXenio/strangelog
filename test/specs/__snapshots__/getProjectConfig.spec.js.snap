// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getProjectConfig when there is a .strangelogrc returns configuration from .strangelogrc merged over defaults 1`] = `
Object {
  "components": Object {
    "comp1": "Comp 1",
    "comp2": "Comp 2",
  },
  "path": "changelog",
}
`;

exports[`getProjectConfig when there is no .strangelogrc returns default configuration 1`] = `
Object {
  "components": Object {},
  "path": "./changelog",
}
`;
